### PR TITLE
feat(icon): add denoify icon

### PR DESCRIPTION
**Pull request by GervinFung**
**Changes proposed:**

- [x] Add
- [ ] Delete
- [ ] Fix
- [ ] Prepare

### CODE REVIEW

**Review by sasial-dev — CHANGES_REQUESTED** [review]
Please respect the icon margins as per our [wiki page](https://github.com/vscode-icons/vscode-icons/wiki/Contributing) on this.

**Comment by GervinFung** [comment]
> Please respect the icon margins as per our [wiki page](https://github.com/vscode-icons/vscode-icons/wiki/Contributing) on this.

Hey some changes was made according to the guideline

File: src/iconsManifest/supportedExtensions.ts
Comment (sasial-dev) [inline src/iconsManifest/supportedExtensions.ts:1314]: This should make your code much eaiser to read:

```suggestion
      extensions: ['.denoifyrc'],
      filenamesGlob: ['.denoifyrc', 'denoify.config'],
      extensionsGlob: ['json', 'yaml', 'yml', 'js', 'cjs'],
```

**Review by sasial-dev — CHANGES_REQUESTED** [review]
Consider also creating a light version of your icon for dark mode users of VSC. Also see my feedback below :)

File: src/iconsManifest/supportedExtensions.ts
Comment (GervinFung) [reply src/iconsManifest/supportedExtensions.ts:1314]: I think that this icon can be used for both dark/light theme, if by creating light icon u mean to create one for light theme then yes, I will do it, otherwise, I think it's fine

**Comment by garronej** [comment]
> Consider also creating a light version of your icon for dark mode users of VSC  

Hi @sasial-dev, the logo has been desined to work both on dark and light mode:  

image

image

What do you think?

**Comment by robertohuertasm** [comment]
@GervinFung, the icon needs to be refactored. It's 42kb and we aim for 2kb max.

**Comment by voidei** [comment]
@garronej Do you happen to have access to a an SVG of the icon? Looking at it in inkscape seems to show that it's simply been vectorized, as it possesses several artifacts (see image) and despite my own best efforts, the most optimized I've managed to get the icon to is 18.73KB - which is still significantly better than what it was previously, but would be ideal to reduce further.

**Comment by garronej** [comment]
@voidei thank you very much for your efforts!  
Unfortunately no, the original logo is a PNG.  
What I did though is literally just superposing the Deno logo to the node logo...

**Comment by voidei** [comment]
@garronej Well, it worked well regardless! Knowing that though, I *should* be able to just do the same thing in inkscape - gonna see what I can cook up.

**Comment by garronej** [comment]
Thank you!!!

**Comment by voidei** [comment]
@garronej Uploaded a few levels of optimisation of it to a [gist](https://gist.github.com/voidei/6d67ca0ab54b948160b26a86bb4f96aa), won't be 100% accurate - and only 32x32 (though given it being an svg, resizing it wouldn't be an issue)

**Comment by garronej** [comment]
Hey @voidei,  
Thank a lot for your effort, svgo is verry cool!  

Unfortunately, the file size is still well above the threshold of 2KB.  

By removing the rain, I managed to decrease the file size to 3.0KB: https://user-images.githubusercontent.com/6702424/219868104-6d064727-d0c4-4fd5-86fa-0e1283ebb3c8.svg  

By removing the dinosaur, I managed to get down to 2.3KB: https://user-images.githubusercontent.com/6702424/219868297-60afe691-79f9-4c6e-84dc-62a296f3448f.svg  

@robertohuertasm is it a no go?

**Comment by voidei** [comment]
@garronej I did a bit more tweaking with it, removed the gradient on the background and merged the dino + eyes - significantly reduced the size down on it to 3.23KB while still retaining the dino and the rain - not sure it's gonna be able to be reduced by much more than that, and I have noticed that most icons lie within the 1-5KB range (depending on the complexity), with both shellcheck and lerna being in excess of 200KB - although that is obviously something you would want to avoid.
file_type_denoify2

**Comment by garronej** [comment]
@voidei Can I have the source of the 3.23KB SVG?  
The one you linked is 11KB.  

And by the way, do yo want to make the change yourself on the branch that opened the PR, so you can get the credit?

**Comment by GervinFung** [comment]
> @garronej I did a bit more tweaking with it, removed the gradient on the background and merged the dino + eyes - significantly reduced the size down on it to 3.23KB while still retaining the dino and the rain - not sure it's gonna be able to be reduced by much more than that, and I have noticed that most icons lie within the 1-5KB range (depending on the complexity), with both shellcheck and lerna being in excess of 200KB - although that is obviously something you would want to avoid. file_type_denoify2

@voidei thank you for the work, we really appreciate it. I have invited both @garronej and @voidei for write access to my repository, so both of you can contribute to the repository

**Comment by voidei** [comment]
@garronej Not sure why that one uploaded to 11KB, but I put the source into a [gist](https://gist.github.com/voidei/9940a8bcbef0fc48adf19b3fd9a11e58) again for you though (and confirmed that it is still only 3KB). And thanks for that @GervinFung, will fix up the optimised icon now 😄

**Comment by garronej** [comment]
Thank you guys!
It's ready for review, I hope the 3.23KB get a pass 🤞🏻

File: src/iconsManifest/supportedExtensions.ts
Comment (voidei) [inline src/iconsManifest/supportedExtensions.ts:1325]: Just wanted to note this, as it'll get picked up by checks. Gotta keep consistent indents 👍 
```suggestion
      extensionsGlob: ['json', 'js', 'cjs'],
```

File: src/iconsManifest/supportedExtensions.ts
Comment (voidei) [inline src/iconsManifest/supportedExtensions.ts:1324]: Gotta add the `.denoifyrc` file into here, otherwise it won't be picked up on by the icons.
```suggestion
      extensions: ['.denoifyrc'],
      filenamesGlob: ['denoify.config'],
```

Comment (voidei) [reply src/iconsManifest/supportedExtensions.ts:1324]: Note: Putting that there instead of doing it myself to ensure @GervinFung still has a high level of contribution to it 👍

File: src/iconsManifest/supportedExtensions.ts
Comment (garronej) [reply src/iconsManifest/supportedExtensions.ts:1325]: Damn sorry about that.  
Good catch

File: src/iconsManifest/supportedExtensions.ts
Comment (garronej) [reply src/iconsManifest/supportedExtensions.ts:1324]: I just wanted to be consistent with the Denoify documentation: https://docs.denoify.land/build-options  

We allow `denoify.config.js` or `denoify.config.json` but thank you if I got confused and made a mistake!

Comment (GervinFung) [reply src/iconsManifest/supportedExtensions.ts:1324]: > Note: Putting that there instead of doing it myself to ensure @GervinFung still has a high level of contribution to it +1

Sorry Im late, just woke up 2 mins ago. Some other stuffs that's happening caught my attention, so wasn't doing much. Ok, u can do it urself too, but anyhow, since u said it, Ill do something with it

**Review by voidei — APPROVED** [review]

**Comment by GervinFung** [comment]
What else is needed before merging?

**Review by robertohuertasm — APPROVED** [review]
LGTM 🚀 Thanks for your contribution ❤️
Finally, we're merging this one! Thank you all!

**Pull request merged**
